src/Payments.sol

```diff
@@ -0,0 +1,572 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./RateChangeQueue.sol";
+
+interface IArbiter {
+    struct ArbitrationResult {
+        bool approved;
```

Comment:
Do we need `approved`? IMO, we should just revert if we're not approving anything, but I'm not really a solidity expert.

Reply:
@Stebalien This is for the Arbitrer contract to gracefully let us know that it is/is not rejecting the payment. The Payment contract does revert the settlement if it sees a disapproval from the arbitrer.

Reply:
The other option is to not gracefully handle any reverts from the `Arbitrer` contract at all and simply revert right away (in Solidity, when contract X calls contract Y and contract Y reverts -> the call to X reverts right away unless you handle it as above).

The way we've implemented it here -> any revert from `Arbitrer` will bubble up right away and the settlement will revert but atleast for payment refusals -> we get to handle it gracefully and do more stuff if we want to before we revert.

Reply:
The Arbitrer SHOULD be able to revert if it wants to.

Reply:
Done. Removed this.